utils/src/test/kotlin/storage/FileArchiverTest.kt

```diff
@@ -22,53 +22,90 @@ package com.here.ort.utils.storage
-        val dir = createTempDir("ort")
+import java.io.File
 
-        fun createFile(path: String) {
-            val file = dir.resolve(path)
-            file.parentFile.safeMkdirs()
-            file.writeText(path)
-        }
+class FileArchiverTest : StringSpec() {
+    lateinit var workingDir: File
+    lateinit var storageDir: File
+    lateinit var storage: LocalFileStorage
+
+    override fun beforeTest(testCase: TestCase) {
+        workingDir = createTempDir("ort").also { it.deleteOnExit() }
+        storageDir = createTempDir("ort").also { it.deleteOnExit() }
+        storage = LocalFileStorage(storageDir)
+    }
 
-        createFile("a")
-        createFile("b")
-        createFile("d/a")
-        createFile("d/b")
+    private fun createFile(path: String) {
+        val file = workingDir.resolve(path)
+        file.parentFile.safeMkdirs()
+        file.writeText(path)
+    }
 
-        val storageDir = createTempDir("ort")
-        val storage = LocalFileStorage(storageDir)
-        val archiver = FileArchiver(listOf("a", "**/a"), storage)
+    init {
+        "All files matching any of the patterns are archived" {
+            createFile("a")
+            createFile("b")
+            createFile("d/a")
+            createFile("d/b")
 
-        archiver.archive(dir, "save")
+            val archiver = FileArchiver(listOf("a", "**/a"), storage)
 
-        val archiveFile = storageDir.resolve("save/archive.zip")
+            archiver.archive(workingDir, "save")
 
-        archiveFile.isFile shouldBe true
+            val archiveFile = storageDir.resolve("save/archive.zip")
 
-        val unzipDir = createTempDir()
-        unzipDir.deleteOnExit()
-        archiveFile.unpack(unzipDir)
+            archiveFile.isFile shouldBe true
 
-        fun assertFileSaved(path: String) {
-            val file = unzipDir.resolve(path)
-            file.isFile shouldBe true
-            file.readText() shouldBe path
-        }
+            val unzipDir = createTempDir()
+            unzipDir.deleteOnExit()
+            archiveFile.unpack(unzipDir)
+
+            fun assertFileSaved(path: String) {
+                val file = unzipDir.resolve(path)
+                file.isFile shouldBe true
+                file.readText() shouldBe path
+            }
 
-        assertFileSaved("a")
-        assertFileSaved("d/a")
+            assertFileSaved("a")
+            assertFileSaved("d/a")
 
-        fun assertFileNotSaved(path: String) {
-            val file = storageDir.resolve("save/$path")
-            file.exists() shouldBe false
+            fun assertFileNotSaved(path: String) {
+                val file = storageDir.resolve("save/$path")
+                file.exists() shouldBe false
+            }
+
+            assertFileNotSaved("b")
+            assertFileNotSaved("d/b")
         }
 
-        assertFileNotSaved("b")
-        assertFileNotSaved("d/b")
+        "All archived files are unarchived" {
+            createFile("a")
```

Comment:
Would it be sufficient to use just two files, e.g. `"a", "c/a"`? .. or even just a single one?

Reply:
I prefer to cover a few more cases here than a single file in case the underlying code is changed.